core/arch/atomic_exports.h

```diff
@@ -723,7 +723,254 @@ atomic_dec_becomes_zero(volatile int *var)
                                  : "r2")
 #        define SET_IF_NOT_ZERO(flag) SET_FLAG(ne, flag)
 #        define SET_IF_NOT_LESS(flag) SET_FLAG(ge, flag)
-#    endif /* X86/ARM */
+#    elif defined(DR_HOST_RISCV64)
+#        define ATOMIC_1BYTE_READ(addr_src, addr_res)               \
+            do {                                                    \
+                __asm__ __volatile__("lbu %0, 0(%1)\n\t"            \
```

Comment:
Is lbu not atomic on RISC-V for byte reads? Why do we need the fence? 

Same for ATOMIC_1BYTE_WRITE.

Reply:
Yes, it is. fence is for preventing re-ordering and fusing in the write buffer. GCC does the same on `atomic_inc(std_bool, bool)`.

Reply:
Is `atomic_load` w/o specifying the model using `memory_order_seq_cst`?  We would want release-acquire semantics.  I'm guessing there's still a barrier/fence of some sort: but is it a different one?

Reply:
GCC's `atomic_load` (without `_explicit`) uses `memory_order_seq_cst` and has 2 fences - before and after the 1byte load. I've implemented atomic load/store functions with acquire/release in mind, hence the fence after the load.